### PR TITLE
Move DP warning suppression to the DataParallel Plugin

### PR DESCRIPTION
**Pull request by ananthsub**
## What does this PR do?

- Simplifies what's under `pytorch_lightning/trainer/`
- Since this is specific to DataParallel, we should include it as part of the plugin definition

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/plugins/training_type/dp.py [hunk 1]
```diff
@@ -22,6 +23,18 @@
 from pytorch_lightning.utilities.apply_func import apply_to_collection
 
 
+def _ignore_scalar_return_in_dp():
```
Comment (awaelchli) [inline pytorch_lightning/plugins/training_type/dp.py:26]: wouldn't it be better to have it where the overrides are, i.e., in pytorch_lightning.overrides?

Comment (carmocca) [reply pytorch_lightning/plugins/training_type/dp.py:26]: I personally prefer it where it is.

Overrides seems more to me for full class/function overrides

**Review by carmocca — APPROVED** [review]
This warning is produced by torch, correct?

File: pytorch_lightning/plugins/training_type/dp.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/plugins/training_type/dp.py:26]: To be clear, the warning is produced by nn.DataParallel which we wrap/override in the files in pl/overrides. The plugin here has nothing to do with that and in fact we have another plugin that uses DataParallel, the DDP2Plugin. So you would have to copy the ignore there too :)

Comment (carmocca) [reply pytorch_lightning/plugins/training_type/dp.py:26]: touché then

Comment (ananthsub) [reply pytorch_lightning/plugins/training_type/dp.py:26]: @awaelchli @carmocca I'll move it to the overrides

**Review by awaelchli — APPROVED** [review]
amazing!

**Pull request merged**